src/frontend/src/catalog/system_catalog/rw_catalog/rw_sources.rs

```diff
@@ -78,7 +79,11 @@ fn read_rw_sources_info(reader: &SysCatalogReaderImpl) -> Result<Vec<RwSource>>
                         .map(|row_encode| row_encode.as_str_name().into()),
                     append_only: source.append_only,
                     connection_id: source.connection_id.map(|id| id as i32),
-                    definition: source.create_sql(),
+                    definition: if source.owner == user_id {
+                        source.create_sql()
+                    } else {
+                        source.redacted_create_sql()
+                    },
```

Comment:
Why do we distinguish owner? I'm not sure whether this is reasonable 🤣

Reply:
User should be able to check secrets in definition somehow.

Reply:
I think one reason user want redaction is that they want to `SHOW CREATE` and share it to us for debugging. For this use case, it seems not very convenient if we distinguish by owner.

@BugenZhao comes up with an idea that maybe we can add sth like `SHOW REDACTD CREATE`. 🤣 (Maybe alternatively we can add a session variable or sth..?)

I'm not sure what users want now. @fuyufjh Do you have any ideas about users' requests? Since I saw you opened #15402

Reply:
BTW on the other hand, force redaction `if source.owner != user_id` sounds good 🤔

Reply:
The major requirement is that users are **concerning** about the secret leaks. Before having a real secret store, I think we should redact for all users.